google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java

```diff
@@ -227,6 +241,12 @@ public Builder setDefinition(TableDefinition definition) {
       return this;
     }
 
+    @Override
+    public Builder setLabels(Map<String, String> labels) {
+      this.labels = ImmutableMap.copyOf(labels);
```

Comment:
ImmutableMap won't work here. We want folks to be able to explicitly set a label to `null`, because this is the way to unset a label when using a `PATCH` request.